sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuShuffleCoalesceExec.scala

```diff
@@ -423,12 +516,55 @@ abstract class CoalesceIteratorBase[T <: AutoCloseable : ClassTag, R](
   override def close(): Unit = {
     serializedTables.forEach(_.close())
```

Comment:
I believe we should group all the autocloseables together, then call `.safeClose` on that sequence. Just in case one of the closes blows up.